Gems/Atom/RHI/DX12/Code/Source/RHI/MemoryView.h

```diff
@@ -77,6 +77,9 @@ namespace AZ
             /// Sets the name of the ID3D12Resource.
             void SetName(const AZStd::string_view& name);
 
+            /// Sets the name of the ID3D12Resource.
+            void SetName(const AZStd::wstring_view& name);
```

Comment:
nit: This might be outside the scope of this PR but `string_view` should really be passed by value and not reference.

There's a useful snippet here from the Abseil guidance that has a good explanation:

> string_view objects are very lightweight, so you should always pass them by value within your methods and functions; don’t pass a const absl::string_view &. (Passing absl::string_view instead of const absl::string_view & has the same algorithmic complexity, but because of register allocation and parameter passing rules, it is generally faster to pass by value in this case.